java-util/src/main/java/io/druid/java/util/common/parsers/RegexParser.java

```diff
@@ -56,12 +59,9 @@ public RegexParser(
       @Override
       public Object apply(String input)
       {
-        final List retVal = Lists.newArrayList(
-            Iterables.transform(
-                listSplitter.split(input),
-                ParserUtils.nullEmptyStringFunction
-            )
-        );
+        final List retVal = StreamSupport.stream(listSplitter.split(input).spliterator(), false)
```

Comment:
If you don't care about perf you could use splitAsList() for shorter code without StreamSupport